### ISSUE TITLE
no-control-regex: Invalid example in documentation doesn't trigger a warning on the demo page

### ISSUE THREAD

**Issue by nfriend** [eslint/eslint#5737]
**What version of ESLint are you using?**
2.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Here's my `.eslintrc.js`:

```javascript
module.exports = {
    "rules": {
        "no-control-regex": "error"
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```javascript
var pattern1 = /\\x1f/;
```

**What did you expect to happen?**
I expected this code to cause a linting error. This code snippet is the first incorrect example in [the documentation](http://eslint.org/docs/rules/no-control-regex).  The second example (`var pattern2 = new RegExp("\x1f");` throws an error as expected.

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
PS C:\temp> eslint .\test.js
PS C:\temp>
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed it.
This rule seems to rely on An old spec of AST.

**Comment by efegurkan** [comment]
Hello,

I did some experiments and have something  at #5943.
